src/createServer.js

```diff
@@ -1,8 +1,41 @@
 'use strict';
 
+const http = require('http');
+const fs = require('fs');
+const url = require('url');
+
 function createServer() {
-  /* Write your code here */
-  // Return instance of http.Server class
+  return http.createServer((req, res) => {
+    const normalizedUrl = new url.URL(req.url, `http://${req.headers.host}`);
+    const { pathname } = normalizedUrl;
+    const fileName = pathname.replace('/file', '') || 'index.html';
```

Comment:
The logic for determining `fileName` assumes that the path will always start with '/file'. If the path is exactly '/file', this will result in an empty string, defaulting to 'index.html'. Consider adding a check to handle this case more explicitly.